### PR TITLE
API – Posts and comments endpoints

### PR DESCRIPTION
**Pull request by KentShikama**
We still need to resolve the scopes discussion. @jhass @jaywink @AugierLe42e @Flaburgan

### CODE REVIEW

File: app/controllers/api/v0/comments_controller.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
+module Api
+  module V0
+    class CommentsController < Api::V0::BaseController
+
```
Comment (diaspora-code-review) [inline app/controllers/api/v0/comments_controller.rb]: Extra empty line detected at class body beginning.

File: spec/integration/api/comments_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    context "valid post ID" do
+      it "succeeds" do
+        post api_v0_post_comments_path(post_id: @status.id), text: "This is a comment", access_token: access_token
+        expect(JSON.parse(response.body)["text"]).to eq("This is a comment")
+      end
+    end
+
+    context "comment too long" do
+      before do
+        post api_v0_post_comments_path(post_id: @status.id), text: "This is a long comment" * 99999, access_token: access_token
```
Comment (diaspora-code-review) [inline spec/integration/api/comments_controller_spec.rb]: Separate every 3 digits in the integer portion of a number with underscores(_).

Comment (diaspora-code-review) [inline spec/integration/api/comments_controller_spec.rb]: Line is too long. [127/120]

File: spec/integration/api/comments_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+      end
+    end
+
+    context "invalid comment ID" do
+      before do
+        post api_v0_post_comments_path(post_id: @status.id), text: "This is a comment", access_token: access_token
+      end
+
+      it "fails to delete" do
+        delete api_v0_post_comment_path(id: 1234567), access_token: access_token
```
Comment (diaspora-code-review) [inline spec/integration/api/comments_controller_spec.rb]: Separate every 3 digits in the integer portion of a number with underscores(_).

**Comment by jhass** [comment]
I also still don't feel too good about reusing the application presenters or at least their json representation. Opposed to a dedicated and stable JSON representation for the API endpoints, maybe through something like jbuilder even.

**Comment by theworldbright** [comment]
See #6636

**Pull request closed without merging**